tests/mesh/write_vec_and_scalar.C

```diff
@@ -0,0 +1,131 @@
+#include "libmesh/exodusII_io.h"
+
+// Ignore unused parameter warnings coming from cppunit headers
+#include <libmesh/ignore_warnings.h>
+#include <cppunit/extensions/HelperMacros.h>
+#include <cppunit/TestCase.h>
+#include <libmesh/restore_warnings.h>
+
+// Basic include files
+#include "libmesh/equation_systems.h"
+#include "libmesh/mesh.h"
+#include "libmesh/mesh_generation.h"
+#include "libmesh/string_to_enum.h"
+#include "libmesh/exodusII_io.h"
+
+// The systems and solvers we may use
+#include "libmesh/diff_solver.h"
+#include "libmesh/steady_solver.h"
+
+#include "mesh/curl_curl_system.h"
+#include "test_comm.h"
+
+// THE CPPUNIT_TEST_SUITE_END macro expands to code that involves
+// std::auto_ptr, which in turn produces -Wdeprecated-declarations
+// warnings.  These can be ignored in GCC as long as we wrap the
+// offending code in appropriate pragmas.  We can't get away with a
+// single ignore_warnings.h inclusion at the beginning of this file,
+// since the libmesh headers pull in a restore_warnings.h at some
+// point.  We also don't bother restoring warnings at the end of this
+// file since it's not a header.
+#include <libmesh/ignore_warnings.h>
+
+// Bring in everything from the libMesh namespace
+using namespace libMesh;
+
+class WriteVecAndScalar : public CppUnit::TestCase
+{
+  /**
+   * This test ensures you can write both vector and scalar variables
+   */
+public:
+  CPPUNIT_TEST_SUITE(WriteVecAndScalar);
+
+  CPPUNIT_TEST(testWrite);
+
+  CPPUNIT_TEST_SUITE_END();
+
+  void testWrite()
+  {
+    Mesh mesh(*TestCommWorld);
+
+    MeshTools::Generation::build_square(
+        mesh, 2, 2, -1., 1., -1., 1., Utility::string_to_enum<ElemType>("TRI6"));
+
+    // Create an equation systems object.
+    EquationSystems equation_systems(mesh);
+
+    // Declare the system "Navier-Stokes" and its variables.
+    CurlCurlSystem & system = equation_systems.add_system<CurlCurlSystem>("CurlCurl");
+
+    // This example only implements the steady-state problem
+    system.time_solver = UniquePtr<TimeSolver>(new SteadySolver(system));
+
+    // Initialize the system
+    equation_systems.init();
+
+    // And the nonlinear solver options
+    DiffSolver & solver = *(system.time_solver->diff_solver().get());
+    solver.quiet = true;
+    solver.verbose = !solver.quiet;
+    solver.max_nonlinear_iterations = 15;
+    solver.relative_step_tolerance = 1.e-3;
+    solver.relative_residual_tolerance = 1.0e-13;
+    solver.absolute_residual_tolerance = 0.0;
+
+    // And the linear solver options
+    solver.max_linear_iterations = 50000;
+    solver.initial_linear_tolerance = 1.e-10;
+
+    system.solve();
```

Comment:
Do we really need to do all this work (assembly/solve/etc) to test this? How about you just write some pre-determined values to the system solution vector, then write and Exodus file and read it back in to make sure they match? I'd prefer not to duplicate all this example code in the unit test (and run a solve!) if possible. What if the solve fails on e.g. Trilinos? That failure would have nothing to do with your actual test...